js/setpassword.js

```diff
@@ -0,0 +1,44 @@
+OC.SetPassword = {
+	init : function() {
+		$('#set-password #submit').click(OC.SetPassword.onClickSetPassword);
+	},
+
+	onClickSetPassword : function(event){
+		event.preventDefault();
+		var passwordObj = $('#password');
+		if (passwordObj.val()){
+			$.post(
+					passwordObj.parents('form').attr('action'),
+					{
+						password : passwordObj.val()
+					},
+					OC.SetPassword._resetDone
+			);
+		}
+	},
+
+	_resetDone : function(result){
+		var resetErrorMsg;
+		if (result){
+			$.post(
+				OC.webroot + '/',
+				{
+					user : window.location.href.split('/').pop(),
+					password : $('#password').val()
+				},
+				function (msg) {
+					OC.redirect(OC.webroot);
+				}
+			);
+		} else {
+			resetErrorMsg = t('core', 'Couldn\'t send reset email. Please contact your administrator.');
```

Comment:
The exception which causes `sendNotificationMail` will cause the code to flow this way.

Reply:
I guess you mean the `setPassword` method of the `UsersController`. Well, then the templates we're sending in that method are being wasted because they'll never be shown.
This is why either you send information via json to be used with ajax, or send templates or redirects for the browser.

I don't know if with the current code it will work properly. I don't think a template response or redirect response work with the `result.status` check. It might be crashing.

Reply:
Yes. I have replaced the mixed responses made earlier with JSONResponse. 3 new fields are added in the setpassword template ( part from the one added earlier ). If the JSONResponse has `result.status` as `error`, then the submit button is also disabled.